src/core/server/execution_context/execution_context_service.test.ts

```diff
@@ -25,78 +25,337 @@ describe('ExecutionContextService', () => {
       service = new ExecutionContextService(core).setup();
     });
 
-    it('sets and gets a value in async context', async () => {
-      const chainA = Promise.resolve().then(async () => {
-        service.set({
-          requestId: '0000',
+    describe('set', () => {
+      it('sets and gets a value in async context', async () => {
+        const chainA = Promise.resolve().then(async () => {
+          service.set({
+            type: 'type-a',
+            name: 'name-a',
+            id: 'id-a',
+            description: 'description-a',
+          });
+          await delay(500);
+          return service.get();
         });
-        await delay(500);
-        return service.get();
+
+        const chainB = Promise.resolve().then(async () => {
+          service.set({
+            type: 'type-b',
+            name: 'name-b',
+            id: 'id-b',
+            description: 'description-b',
+          });
+          await delay(100);
+          return service.get();
+        });
+
+        expect(
+          await Promise.all([chainA, chainB]).then((results) =>
+            results.map((result) => result?.toJSON())
+          )
+        ).toEqual([
+          {
+            type: 'type-a',
+            name: 'name-a',
+            id: 'id-a',
+            description: 'description-a',
+            parent: undefined,
+          },
+
+          {
+            type: 'type-b',
+            name: 'name-b',
+            id: 'id-b',
+            description: 'description-b',
+            parent: undefined,
+          },
+        ]);
       });
 
-      const chainB = Promise.resolve().then(async () => {
-        service.set({
-          requestId: '1111',
+      it('a sequentual call rewrites the context', async () => {
+        const result = await Promise.resolve().then(async () => {
+          service.set({
+            type: 'type-a',
+            name: 'name-a',
+            id: 'id-a',
+            description: 'description-a',
+          });
+          service.set({
+            type: 'type-b',
+            name: 'name-b',
+            id: 'id-b',
+            description: 'description-b',
+          });
+
+          return service.get();
+        });
+
+        expect(result?.toJSON()).toEqual({
+          type: 'type-b',
+          name: 'name-b',
+          id: 'id-b',
+          description: 'description-b',
+          parent: undefined,
```

Comment:
I'm curious if this is the behavior we actually want instead of subsequent calls using the previous context as the parent. If `pluginA` calls a sync API on `pluginB` and they both set context, wouldn't we want to track that relationship?

Reply:
> If pluginA calls a sync API on pluginB and they both set context, wouldn't we want to track that relationship?

`set` is Core-specific API which won't be provided to plugins. `withContext` which is exposed to plugins,  supports context stacking https://github.com/elastic/kibana/blob/ae4571973e22244f7680d672f944a6476a18bd80/src/core/server/execution_context/execution_context_service.test.ts#L231